11-deny-egress-traffic-from-an-application.md

```diff
@@ -83,11 +83,18 @@ spec:
   - Egress
   egress:
   # allow DNS resolution
-  - ports:
-    - port: 53
-      protocol: UDP
-    - port: 53
-      protocol: TCP
+  - to:
+    - namespaceSelector:
+        matchLabels:
+          kubernetes.io/metadata.name: kube-system
+    - podSelector:
```

Comment:
This is not going doing what you expect it to do. The syntax here will be interpreted into an OR. AKA namespaces with label kubernetes.io/metadata.name: kube-system or pods with label k8s-app: kube-dns.

What you need is an AND and for that you have to remove the - from the podSelector part.

More details can be found here https://kubernetes.io/docs/concepts/services-networking/network-policies/#behavior-of-to-and-from-selectors.

Also we need to adjust the sentence above this paragraph "So we slightly modify the YAML file to allow all outbound traffic on DNS ports (53/udp and 53/tcp):" to explain what the policy does